packages/jellyfish-api-core/src/category/icxorderbook.ts

```diff
@@ -0,0 +1,147 @@
+import { BigNumber, ApiClient } from '../.'
+
+/**
+ * icxorderbook RPCs for DeFi Blockchain
+ */
+export class ICXOrderBook {
+  private readonly client: ApiClient
+
+  constructor (client: ApiClient) {
+    this.client = client
+  }
+
+  /**
+   * Create and submits an ICX order creation transaction.
+   *
+   * @param {ICXOrder} order
+   * @param {string} [order.tokenFrom] Symbol or id of selling token
+   * @param {string} [order.chainFrom] Symbol or id of selling chain
+   * @param {string} [order.chainTo] Symbol or id of buying chain
+   * @param {string} [order.tokenTo] Symbol or id of buying token
+   * @param {string} [order.ownerAddress] Address of DFI token for fees and selling tokens in case of DFC/BTC order type
+   * @param {string} [order.receivePubkey] pubkey which can claim external HTLC in case of EXT/DFC order type
+   * @param {BigNumber} [order.amountFrom] tokenFrom coins amount
+   * @param {BigNumber} [order.orderPrice] Price per unit
+   * @param {number} [order.expiry] Number of blocks until the order expires, default 2880 DFI blocks
+   * @param {InputUTXO[]} inputUTXOs Specific utxos to spend
+   * @param {string} [inputUTXOs.txid] transaction Id
+   * @param {number} [inputUTXOs.vout] The output number
+   * @return {Promise<ICXGenericResult>} Object indluding transaction id of the the result transaction
+   */
+  async createOrder (order: ICXOrder, inputUTXOs: InputUTXO[] = []): Promise<ICXGenericResult> {
+    return await this.client.call(
+      'icx_createorder',
+      [
+        order, inputUTXOs
+      ],
+      'bignumber'
+    )
+  }
+}
+
+export enum ICXOrderStatus {
+  OPEN = 'OPEN',
+  CLOSED = 'CLOSED',
+  FILLED = 'FILLED',
+  EXPIRED = 'EXPIRED'
+}
+
+export enum ICXOrderType {
+  INTERNAL = 'INTERNAL',
+  EXTERNAL = 'EXTERNAL',
+}
+
+/** ICX order */
+export interface ICXOrder {
+  /** Symbol or id of selling token */
+  tokenFrom?: string
+  /** Symbol or id of selling chain */
+  chainFrom?: string
+  /** Symbol or id of buying chain */
+  chainTo?: string
+  /** Symbol or id of buying token */
+  tokenTo?: string
+  /** Address of DFI token for fees and selling tokens in case of DFC/BTC order type */
+  ownerAddress?: string
+  /** pubkey which can claim external HTLC in case of EXT/DFC order type */
+  // NOTE(surangap): c++ side this as number, but no type checks done. should be corrected from c++ side?
+  receivePubkey?: string
+  /** tokenFrom coins amount */
+  amountFrom: BigNumber
+  /** Price per unit */
+  orderPrice: BigNumber
+  /** Number of blocks until the order expires, default 2880 DFI blocks */
+  expiry?: number
+}
+
+/** Input UTXO */
+export interface InputUTXO {
```

Comment:
```suggestion
export interface UTXO {
```

Call the UTXO as it's universally called UTXO everywhere in `jellyfish-api-core`